test/test_utils.py

```diff
@@ -259,6 +259,19 @@ def run_fn(input):
 
             self.assertEqual(grad_with_checkpointing, grad_no_checkpointing)
 
+    @unittest.skipIf(not HAS_CUDA, 'No CUDA')
+    def test_checkpoint_not_preserve_rng_state_and_without_reentrant(self):
+        inp = torch.randn(20000, device='cuda').requires_grad_()
```

Comment:
Could you make that input of size 2 please? no need for it to be so big!

Reply:
Fair enough :) will do